packages/core/src/publicCredential/PublicCredential.chain.ts

```diff
@@ -147,31 +168,40 @@ export async function credentialFromChain(
     )
   }
 
-  if (!api.tx.did.submitDidCall.is(extrinsic)) {
+  if (!isBatch(api, extrinsic) && !api.tx.did.submitDidCall.is(extrinsic)) {
     throw new SDKErrors.PublicCredentialError(
-      'Extrinsic should be a did.submitDidCall extrinsic'
+      'Extrinsic should be either a `did.submitDidCall` extrinsic or a batch with at least a `did.submitDidCall` extrinsic'
     )
   }
 
-  const extrinsicCalls = flattenCalls(api, extrinsic.args[0].call)
-  const extrinsicDidOrigin = didFromChain(extrinsic.args[0].did)
+  // If we're dealing with a batch, flatten any nested `submit_did_call` calls,
+  // otherwise the extrinsic is itself a submit_did_call, so just take it.
+  const didCalls = isBatch(api, extrinsic)
+    ? extrinsic.args[0].flatMap((batchCall) =>
+        extractDidCallsFromBatchCall(api, batchCall)
+      )
+    : [extrinsic]
 
-  const credentialCreationCalls = extrinsicCalls.filter(
-    (call): call is GenericCall<typeof api.tx.publicCredentials.add.args> =>
-      api.tx.publicCredentials.add.is(call)
-  )
+  // From the list of DID calls, only consider public_credentials::add calls, bundling each of them with their DID submitter.
```

Comment:
I was frivolous with comments during refactoring, please confirm that they still fit